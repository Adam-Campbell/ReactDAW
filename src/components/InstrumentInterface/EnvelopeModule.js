import React from 'react';
import PropTypes from 'prop-types';
import EnhancedSelectInput from '../EnhancedSelectInput';
import EnhancedRangeInput from '../EnhancedRangeInput';

const EnvelopeModule = props => (
    <div className="instrument-interface__module-container">
        <h2 className="instrument-interface__module-heading">Envelope</h2>
        <EnhancedRangeInput 
            inputId={'envelope-attack'}
            label={'Attack'}
            min={0.005}
            max={1}
            step={0.005}
            value={props.envelopeData.attack}
            handleChange={props.handleChange}
            identifier={props.instrumentId}
            propertyPathArray={[...props.additionalNesting, 'envelope', 'attack']}
        />
        <EnhancedSelectInput
            inputId={'envelope-attack-curve'}
            label={'Attack Curve'}
            value={props.envelopeData.attackCurve}
            handleChange={props.handleChange}
            identifier={props.instrumentId}
            propertyPathArray={[...props.additionalNesting, 'envelope', 'attackCurve']}
            options={[
                {value: 'linear', text: 'Linear'},
                {value: 'exponential', text: 'Exponential'},
                {value: 'sine', text: 'Sine'},
                {value: 'cosine', text: 'Cosine'},
                {value: 'bounce', text: 'Bounce'},
                {value: 'ripple', text: 'Ripple'},
                {value: 'step', text: 'Step'}
            ]}
        />
        <EnhancedRangeInput 
            inputId={'envelope-decay'}
            label={'Decay'}
            min={0.005}
            max={1}
            step={0.005}
            value={props.envelopeData.decay}
            handleChange={props.handleChange}
            identifier={props.instrumentId}
            propertyPathArray={[...props.additionalNesting, 'envelope', 'decay']}
        />
        <EnhancedRangeInput 
            inputId={'envelope-release'}
            label={'Release'}
            min={0.005}
            max={1}
            step={0.005}
            value={props.envelopeData.release}
            handleChange={props.handleChange}
            identifier={props.instrumentId}
            propertyPathArray={[...props.additionalNesting, 'envelope', 'release']}
        />
        <EnhancedSelectInput
            inputId={'envelope-release-curve'}
            label={'Release Curve'}
            value={props.envelopeData.releaseCurve}
            handleChange={props.handleChange}
            identifier={props.instrumentId}
            propertyPathArray={[...props.additionalNesting, 'envelope', 'releaseCurve']}
            options={[
                {value: 'linear', text: 'Linear'},
                {value: 'exponential', text: 'Exponential'},
                {value: 'sine', text: 'Sine'},
                {value: 'cosine', text: 'Cosine'},
                {value: 'bounce', text: 'Bounce'},
                {value: 'ripple', text: 'Ripple'},
                {value: 'step', text: 'Step'}
            ]}

        />
        <EnhancedRangeInput 
            inputId={'envelope-sustain'}
            label={'Sustain'}
            min={0}
            max={1}
            step={0.005}
            value={props.envelopeData.sustain}
            handleChange={props.handleChange}
            identifier={props.instrumentId}
            propertyPathArray={[...props.additionalNesting, 'envelope', 'sustain']}
        />
    </div>
);

EnvelopeModule.propTypes = {
    envelopeData: PropTypes.object.isRequired,
    instrumentId: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
    additionalNesting: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default EnvelopeModule;