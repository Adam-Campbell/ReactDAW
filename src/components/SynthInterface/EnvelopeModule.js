import React from 'react';
import RangeInput from './RangeInput';
import SelectInput from './SelectInput';

const EnvelopeModule = props => (
    <div className="synth-interface__module-container">
        <h2 className="synth-interface__module-heading">Envelope</h2>
        <RangeInput 
            inputId={'envelope-attack'}
            label={'Attack'}
            min={0.005}
            max={1}
            step={0.005}
            value={props.envelopeData.attack}
            handleChange={props.handleChange}
            instrumentId={props.instrumentId}
            propertyPathArray={['envelope', 'attack']}
        />
        <SelectInput
            inputId={'envelope-attack-curve'}
            label={'Attack Curve'}
            value={props.envelopeData.attackCurve}
            handleChange={props.handleChange}
            instrumentId={props.instrumentId}
            propertyPathArray={['envelope', 'attackCurve']}
            options={[
                {value: 'linear', text: 'Linear'},
                {value: 'exponential', text: 'Exponential'},
                {value: 'sine', text: 'Sine'},
                {value: 'cosine', text: 'Cosine'},
                {value: 'bounce', text: 'Bounce'},
                {value: 'ripple', text: 'Ripple'},
                {value: 'step', text: 'Step'}
            ]}
        />
        <RangeInput 
            inputId={'envelope-decay'}
            label={'Decay'}
            min={0.005}
            max={1}
            step={0.005}
            value={props.envelopeData.decay}
            handleChange={props.handleChange}
            instrumentId={props.instrumentId}
            propertyPathArray={['envelope', 'decay']}
        />
        <RangeInput 
            inputId={'envelope-release'}
            label={'Release'}
            min={0.005}
            max={1}
            step={0.005}
            value={props.envelopeData.release}
            handleChange={props.handleChange}
            instrumentId={props.instrumentId}
            propertyPathArray={['envelope', 'release']}
        />
        <SelectInput
            inputId={'envelope-release-curve'}
            label={'Release Curve'}
            value={props.envelopeData.releaseCurve}
            handleChange={props.handleChange}
            instrumentId={props.instrumentId}
            propertyPathArray={['envelope', 'releaseCurve']}
            options={[
                {value: 'linear', text: 'Linear'},
                {value: 'exponential', text: 'Exponential'},
                {value: 'sine', text: 'Sine'},
                {value: 'cosine', text: 'Cosine'},
                {value: 'bounce', text: 'Bounce'},
                {value: 'ripple', text: 'Ripple'},
                {value: 'step', text: 'Step'}
            ]}
        />
        <RangeInput 
            inputId={'envelope-sustain'}
            label={'Sustain'}
            min={0}
            max={1}
            step={0.005}
            value={props.envelopeData.sustain}
            handleChange={props.handleChange}
            instrumentId={props.instrumentId}
            propertyPathArray={['envelope', 'sustain']}
        />
    </div>
);

export default EnvelopeModule;