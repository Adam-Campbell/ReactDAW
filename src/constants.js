export const synthTypes = {
    default: 'synth',
    am: 'amSynth',
    fm: 'fmSynth',
    duo: 'duoSynth',
    mono: 'monoSynth'
};

export const UIColors = {
    deepPurple: '#201826',
    lightPurple: '#47426c',
    pink: '#d86597',
    offWhite: '#e0e0e0',
    brightBlue: '#00bff3',
    brightYellow: '#f4ab54',
    brightRed: '#e23340',
    brightGreen: '#25e452'
};

export const effectTypes = {
    autoFilter: 'autoFilter',
    bitCrusher: 'bitCrusher',
    chorus: 'chorus',
    compressor: 'compressor',
    distortion: 'distortion',
    eq3: 'eq3',
    feedbackDelay: 'feedbackDelay',
    filter: 'filter',
    freeverb: 'freeverb',
    gate: 'gate',
    jcReverb: 'jcReverb',
    lfo: 'lfo',
    limiter: 'limiter',
    multibandCompressor: 'multibandCompressor',
    phaser: 'phaser',
    pingPongDelay: 'pingPongDelay',
    pitchShift: 'pitchShift',
    tremolo: 'tremolo',
    vibrato: 'vibrato'
};

export const defaultSynthData = {
    envelope: {
        attack: 0.005,
        attackCurve: 'linear',
        decay: 0.1,
        release: 1,
        releaseCurve: 'exponential',
        sustain: 0.3
    },
    oscillator: {
        detune: 0,
        mute: false,
        phase: 0,
        type: 'triangle',
        volume: 0
    },
    portamento: 0,
    volume: 0
}

export const synthData = {
    synth: {
        envelope: {
            attack: 0.005,
            attackCurve: 'linear',
            decay: 0.1,
            release: 1,
            releaseCurve: 'exponential',
            sustain: 0.3
        },
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    amSynth: {
        detune: 0,
        envelope: {
            attack: 0.01,
            attackCurve: 'linear',
            decay: 0.01,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        harmonicity: 3,
        modulation: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
        },
        modulationEnvelope: {
            attack: 0.5,
            attackCurve: 'linear',
            decay: 0,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    fmSynth: {
        detune: 0,
        envelope: {
            attack: 0.01,
            attackCurve: 'linear',
            decay: 0.01,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        harmonicity: 3,
        modulation: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
        },
        modulationEnvelope: {
            attack: 0.5,
            attackCurve: 'linear',
            decay: 0,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        modulationIndex: 10,
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    duoSynth: {
        harmonicity: 1.5,
        portamento: 0,
        vibratoAmount: 0.5,
        vibratoRate: 5,
        voice0: {
            detune: 0,
            envelope: {
                attack: 0.01,
                attackCurve: 'linear',
                decay: 0,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            filter: {
                Q: 6,
                frequency: 350,
                gain: 0,
                rolloff: -24,
                type: 'lowpass'
            },
            filterEnvelope: {
                attack: 0.01,
                attackCurve: 'linear',
                baseFrequency: 200,
                decay: 0,
                exponent: 2,
                octaves: 7,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            oscillator: {
                detune: 0,
                mute: false,
                phase: 0,
                type: 'sine',
                volume: 0
            },
            portamento: 0,
            volume: -9.999999999999998
        },
        voice1: {
            detune: 0,
            envelope: {
                attack: 0.01,
                attackCurve: 'linear',
                decay: 0,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            filter: {
                Q: 6,
                frequency: 350,
                gain: 0,
                rolloff: -24,
                type: 'lowpass'
            },
            filterEnvelope: {
                attack: 0.01,
                attackCurve: 'linear',
                baseFrequency: 200,
                decay: 0,
                exponent: 2,
                octaves: 7,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            oscillator: {
                detune: 0,
                mute: false,
                phase: 0,
                type: 'sine',
                volume: 0
            },
            portamento: 0,
            volume: -9.999999999999998
        },
        volume: 0
    },
    monoSynth: {
        detune: 0,
        envelope: {
            attack: 0.005,
            attackCurve: 'linear',
            decay: 0.1,
            release: 1,
            releaseCurve: 'exponential',
            sustain: 0.9
        },
        filter: {
            Q: 6,
            frequency: 350,
            gain: 0,
            rolloff: -24,
            type: 'lowpass'
        },
        filterEnvelope: {
            attack: 0.06,
            attackCurve: 'linear',
            baseFrequency: 200,
            decay: 0.2,
            exponent: 2,
            octaves: 7,
            release: 2,
            releaseCurve: 'exponential',
            sustain: 0.5
        },
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
        },
        portamento: 0,
        volume: 0
    }
};


export const effectData = {
    autoFilter: {
        baseFrequency: 200,
        depth: 1,
        filter: {
            Q: 1,
            gain: 0,
            rolloff: -12,
            type: 'lowpass'
        },
        frequency: 1,
        octaves: 2.6,
        type: 'sine',
        wet: 1
    },
    bitCrusher: {
        bits: 4,
        wet: 1
    },
    chorus: {
        delayTime: 3.5,
        depth: 0.7,
        frequency: 1.5,
        spread: 180,
        type: 'sine',
        wet: 1
    },
    compressor: {
        attack: 0.003,
        knee: 30,
        ratio: 12,
        release: 0.25,
        threshold: -24
    },
    distortion: {
        distortion: 0.4,
        oversample: 'none',
        wet: 1
    },
    eq3: {
        high: 0,
        highFrequency: 2500,
        low: 0,
        lowFrequency: 400,
        mid: 0
    },
    feedbackDelay: {
        delayTime: 0.25,
        feedback: 0.125,
        wet: 1
    },
    filter: {
        Q: 1,
        frequency: 350,
        gain: 0,
        rolloff: -12,
        type: 'lowpass'
    },
    freeverb: {
        dampening: 3000,
        roomSize: 0.7,
        wet: 1
    },
    gate: {
        attack: 0.1, 
        release: 0.1,
        threshold: -39.99999999999999
    },
    jcReverb: {
        roomSize: 0.5,
        wet: 1,
    },
    lfo: {
        amplitude: 1,
        frequency: 2,
        max: 1,
        min: 0,
        phase: 0,
        type: 'sine',
        units: 'number'
    },
    limiter: {
        threshold: -12
    },
    multibandCompressor: {
        high: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        },
        highFrequency: 2500,
        low: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        },
        lowFrequency: 400,
        mid: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        }
    },
    phaser: {
        Q: 10,
        baseFrequency: 350,
        frequency: 0.5,
        octaves: 3,
        wet: 1
    },
    pingPongDelay: {
        delayTime: 0.25,
        wet: 1
    },
    pitchShift: {
        delayTime: 0,
        feedBack: 0,
        pitch: 0,
        wet: 1,
        windowSize: 0.1
    },
    tremolo: {
        depth: 0.5,
        frequency: 10,
        spread: 180,
        type: 'sine',
        wet: 1
    },
    vibrato: {
        depth: 0.1,
        frequency: 5,
        type: 'sine',
        wet: 1
    }
}