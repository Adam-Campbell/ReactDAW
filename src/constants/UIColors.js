// export const UIColors = {
//     deepPurple: '#030303',
//     lightPurple: '#131313',
//     pink: '#23af62',
//     offWhite: '#e0e0e0',
//     brightBlue: '#00bff3',
//     brightYellow: '#f4ab54',
//     brightRed: '#e23340',
//     brightGreen: '#25e452',
//     darkGrey: '#222222'
// };

export const UIColors = {
    main: '#23af62',
    mainAlt: '#1d9653',
    backgroundDark: '#030303',
    backgroundMedium: '#131313',
    backgroundLight: '#232323',
    offWhite: '#e0e0e0',
    offWhiteAlt: '#ccc9c9',
    brightBlue: '#00bff3',
    brightRed: '#e23340',
    brightYellow: '#f4ab54',
    pink: '#d86597'
}

// background gradient overlay: linear-gradient(transparent, 60%, #0b0b0b);

// medium old

// export const UIColors = {
//     deepPurple: '#14161c',
//     lightPurple: '#252f48',
//     pink: '#aa9c9c',
//     offWhite: '#e0e0e0',
//     brightBlue: '#00bff3',
//     brightYellow: '#f4ab54',
//     brightRed: '#e23340',
//     brightGreen: '#25e452',
//     darkGrey: '#222222'
// };


// oldest old

// export const UIColors = {
//     deepPurple: '#201826',
//     lightPurple: '#47426c',
//     pink: '#d86597',
//     offWhite: '#e0e0e0',
//     brightBlue: '#00bff3',
//     brightYellow: '#f4ab54',
//     brightRed: '#e23340',
//     brightGreen: '#25e452',
//     darkGrey: '#222222'
// };
