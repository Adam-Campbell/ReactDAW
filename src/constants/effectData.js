export const effectData = {
    autoFilter: {
        baseFrequency: 200,
        depth: 1,
        filter: {
            Q: 1,
            gain: 0,
            rolloff: -12,
            type: 'lowpass'
        },
        frequency: 1,
        octaves: 2.6,
        type: 'sine',
        wet: 1
    },
    bitCrusher: {
        bits: 4,
        wet: 1
    },
    chorus: {
        delayTime: 3.5,
        depth: 0.7,
        frequency: 1.5,
        spread: 180,
        type: 'sine',
        wet: 1
    },
    compressor: {
        attack: 0.003,
        knee: 30,
        ratio: 12,
        release: 0.25,
        threshold: -24
    },
    distortion: {
        distortion: 0.4,
        oversample: 'none',
        wet: 1
    },
    eq3: {
        high: 0,
        highFrequency: 2500,
        low: 0,
        lowFrequency: 400,
        mid: 0
    },
    feedbackDelay: {
        delayTime: '8n',
        feedback: 0.125,
        wet: 1
    },
    filter: {
        Q: 1,
        frequency: 350,
        gain: 0,
        rolloff: -12,
        type: 'lowpass'
    },
    freeverb: {
        dampening: 3000,
        roomSize: 0.7,
        wet: 1
    },
    gate: {
        attack: 0.1, 
        release: 0.1,
        threshold: -39.99999999999999
    },
    jcReverb: {
        roomSize: 0.5,
        wet: 1,
    },
    lfo: {
        amplitude: 1,
        frequency: 2,
        max: 1,
        min: 0,
        phase: 0,
        type: 'sine',
        units: 'number'
    },
    limiter: {
        threshold: -12
    },
    multibandCompressor: {
        high: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        },
        highFrequency: 2500,
        low: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        },
        lowFrequency: 400,
        mid: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        }
    },
    phaser: {
        Q: 10,
        baseFrequency: 350,
        frequency: 0.5,
        octaves: 3,
        wet: 1
    },
    pingPongDelay: {
        delayTime: 0.25,
        wet: 1
    },
    pitchShift: {
        delayTime: 0,
        feedBack: 0,
        pitch: 0,
        wet: 1,
        windowSize: 0.1
    },
    tremolo: {
        depth: 0.5,
        frequency: 10,
        spread: 180,
        type: 'sine',
        wet: 1
    },
    vibrato: {
        depth: 0.1,
        frequency: 5,
        type: 'sine',
        wet: 1
    }
};
