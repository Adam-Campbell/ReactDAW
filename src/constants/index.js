export * from './UIColors';
export * from './programData';
export * from './instrumentData';
export * from './effectData';