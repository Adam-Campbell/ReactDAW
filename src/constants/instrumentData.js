// export const drumSampleURLs = {
//     kick001: 'drum-samples/kicks/kick_001.mp3',
//     kick002: 'drum-samples/kicks/kick_002.mp3',
//     kick003: 'drum-samples/kicks/kick_003.mp3',
//     kick004: 'drum-samples/kicks/kick_004.mp3',
//     kick005: 'drum-samples/kicks/kick_005.mp3',
//     kick006: 'drum-samples/kicks/kick_006.mp3',
//     kick007: 'drum-samples/kicks/kick_007.mp3',
//     kick008: 'drum-samples/kicks/kick_008.mp3',
//     kick009: 'drum-samples/kicks/kick_009.mp3',
//     kick010: 'drum-samples/kicks/kick_010.mp3',
//     snare001: 'drum-samples/snares/snare_001.mp3',
//     snare002: 'drum-samples/snares/snare_002.mp3',
//     snare003: 'drum-samples/snares/snare_003.mp3',
//     snare004: 'drum-samples/snares/snare_004.mp3',
//     snare005: 'drum-samples/snares/snare_005.mp3',
//     snare006: 'drum-samples/snares/snare_006.mp3',
//     snare007: 'drum-samples/snares/snare_007.mp3',
//     clap001: 'drum-samples/claps/clap_001.mp3',
//     clap002: 'drum-samples/claps/clap_002.mp3',
//     clap003: 'drum-samples/claps/clap_003.mp3',
//     clap004: 'drum-samples/claps/clap_004.mp3',
//     closedHat001: 'drum-samples/closed-hats/closed-hat_001.mp3',
//     closedHat002: 'drum-samples/closed-hats/closed-hat_002.mp3',
//     closedHat003: 'drum-samples/closed-hats/closed-hat_003.mp3',
//     closedHat004: 'drum-samples/closed-hats/closed-hat_004.mp3',
//     closedHat005: 'drum-samples/closed-hats/closed-hat_005.mp3',
//     closedHat006: 'drum-samples/closed-hats/closed-hat_006.mp3',
//     openHat001: 'drum-samples/open-hats/open-hat_001.mp3',
//     openHat002: 'drum-samples/open-hats/open-hat_002.mp3',
//     openHat003: 'drum-samples/open-hats/open-hat_003.mp3',
//     openHat004: 'drum-samples/open-hats/open-hat_004.mp3',
//     openHat005: 'drum-samples/open-hats/open-hat_005.mp3',
//     crash001: 'drum-samples/crashes/crash_001.mp3',
//     crash002: 'drum-samples/crashes/crash_002.mp3',
//     crash003: 'drum-samples/crashes/crash_003.mp3',
//     crash004: 'drum-samples/crashes/crash_004.mp3',
//     ride001: 'drum-samples/rides/ride_001.mp3',
//     ride002: 'drum-samples/rides/ride_002.mp3',
//     ride003: 'drum-samples/rides/ride_003.mp3',
//     highTom001: 'drum-samples/high-toms/high-tom_001.mp3',
//     highTom002: 'drum-samples/high-toms/high-tom_002.mp3',
//     highTom003: 'drum-samples/high-toms/high-tom_003.mp3',
//     midTom001: 'drum-samples/mid-toms/mid-tom_001.mp3',
//     midTom002: 'drum-samples/mid-toms/mid-tom_002.mp3',
//     midTom003: 'drum-samples/mid-toms/mid-tom_003.mp3',
//     lowTom001: 'drum-samples/low-toms/low-tom_001.mp3',
//     lowTom002: 'drum-samples/low-toms/low-tom_002.mp3',
//     fx001: 'drum-samples/fx/fx_001.mp3',
//     fx002: 'drum-samples/fx/fx_002.mp3',
//     fx003: 'drum-samples/fx/fx_003.mp3',
//     fx004: 'drum-samples/fx/fx_004.mp3'
// };

import drumSampleURLs from '../soundFiles';

export const instrumentData = {
    synth: {
        envelope: {
            attack: 0.005,
            attackCurve: 'linear',
            decay: 0.1,
            release: 1,
            releaseCurve: 'exponential',
            sustain: 0.3
        },
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    amSynth: {
        detune: 0,
        envelope: {
            attack: 0.01,
            attackCurve: 'linear',
            decay: 0.01,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        harmonicity: 3,
        modulation: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
        },
        modulationEnvelope: {
            attack: 0.5,
            attackCurve: 'linear',
            decay: 0,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    fmSynth: {
        detune: 0,
        envelope: {
            attack: 0.01,
            attackCurve: 'linear',
            decay: 0.01,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        harmonicity: 3,
        modulation: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
        },
        modulationEnvelope: {
            attack: 0.5,
            attackCurve: 'linear',
            decay: 0,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        modulationIndex: 10,
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    duoSynth: {
        harmonicity: 1.5,
        portamento: 0,
        vibratoAmount: 0.5,
        vibratoRate: 5,
        voice0: {
            detune: 0,
            envelope: {
                attack: 0.01,
                attackCurve: 'linear',
                decay: 0,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            filter: {
                Q: 6,
                frequency: 350,
                gain: 0,
                rolloff: -24,
                type: 'lowpass'
            },
            filterEnvelope: {
                attack: 0.01,
                attackCurve: 'linear',
                baseFrequency: 200,
                decay: 0,
                exponent: 2,
                octaves: 7,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            oscillator: {
                detune: 0,
                mute: false,
                phase: 0,
                type: 'sine',
                volume: 0
            },
            portamento: 0,
            volume: -9.999999999999998
        },
        voice1: {
            detune: 0,
            envelope: {
                attack: 0.01,
                attackCurve: 'linear',
                decay: 0,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            filter: {
                Q: 6,
                frequency: 350,
                gain: 0,
                rolloff: -24,
                type: 'lowpass'
            },
            filterEnvelope: {
                attack: 0.01,
                attackCurve: 'linear',
                baseFrequency: 200,
                decay: 0,
                exponent: 2,
                octaves: 7,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            oscillator: {
                detune: 0,
                mute: false,
                phase: 0,
                type: 'sine',
                volume: 0
            },
            portamento: 0,
            volume: -9.999999999999998
        },
        volume: 0
    },
    monoSynth: {
        detune: 0,
        envelope: {
            attack: 0.005,
            attackCurve: 'linear',
            decay: 0.1,
            release: 1,
            releaseCurve: 'exponential',
            sustain: 0.9
        },
        filter: {
            Q: 6,
            frequency: 350,
            gain: 0,
            rolloff: -24,
            type: 'lowpass'
        },
        filterEnvelope: {
            attack: 0.06,
            attackCurve: 'linear',
            baseFrequency: 200,
            decay: 0.2,
            exponent: 2,
            octaves: 7,
            release: 2,
            releaseCurve: 'exponential',
            sustain: 0.5
        },
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    drumKit: {
        kick: {
            sample: drumSampleURLs.kick001,
            volume: 0,
            pan: 0
        },
        snare: {
            sample: drumSampleURLs.snare001,
            volume: 0,
            pan: 0
        },
        openHat: {
            sample: drumSampleURLs.openHat001,
            volume: 0,
            pan: 0
        },
        closedHat: {
            sample: drumSampleURLs.closedHat001,
            volume: 0,
            pan: 0
        },
        clap: {
            sample: drumSampleURLs.clap001,
            volume: 0,
            pan: 0
        },
        crash: {
            sample: drumSampleURLs.crash001,
            volume: 0,
            pan: 0
        },
        ride: {
            sample: drumSampleURLs.ride001,
            volume: 0,
            pan: 0
        },
        highTom: {
            sample: drumSampleURLs.highTom001,
            volume: 0,
            pan: 0
        },
        midTom: {
            sample: drumSampleURLs.midTom001,
            volume: 0,
            pan: 0
        },
        lowTom: {
            sample: drumSampleURLs.lowTom001,
            volume: 0,
            pan: 0
        },
        fx: {
            sample: drumSampleURLs.fx001,
            volume: 0,
            pan: 0
        }
    }
};
