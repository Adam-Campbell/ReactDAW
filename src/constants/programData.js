export const modalTypes = {
    open: 'Open',
    saveAs: 'Save As',
    help: 'help'
};

export const toolTypes = {
    cursor: 'cursor',
    pencil: 'pencil',
    selection: 'selection'
};

export const snapSettings = {
    _32t: '32t',
    _32n: '32n',
    _16t: '16t',
    _16n: '16n',
    _8t: '8t',
    _8n: '8n',
    _4t: '4t',
    _4n: '4n',
    _2t: '2t',
    _2n: '2n',
    _1m: '1m'
};

export const noteDurationSettings = {
    _32t: '32t',
    _32n: '32n',
    _16t: '16t',
    _16n: '16n',
    _8t: '8t',
    _8n: '8n',
    _4t: '4t',
    _4n: '4n',
    _2t: '2t',
    _2n: '2n',
    _1m: '1m'
};

export const instrumentTypes = {
    default: 'synth',
    am: 'amSynth',
    fm: 'fmSynth',
    duo: 'duoSynth',
    mono: 'monoSynth',
    drumKit: 'drumKit'
};

export const effectTypes = {
    autoFilter: 'autoFilter',
    bitCrusher: 'bitCrusher',
    chorus: 'chorus',
    compressor: 'compressor',
    distortion: 'distortion',
    eq3: 'eq3',
    feedbackDelay: 'feedbackDelay',
    filter: 'filter',
    freeverb: 'freeverb',
    gate: 'gate',
    jcReverb: 'jcReverb',
    lfo: 'lfo',
    limiter: 'limiter',
    multibandCompressor: 'multibandCompressor',
    phaser: 'phaser',
    pingPongDelay: 'pingPongDelay',
    pitchShift: 'pitchShift',
    tremolo: 'tremolo',
    vibrato: 'vibrato'
};