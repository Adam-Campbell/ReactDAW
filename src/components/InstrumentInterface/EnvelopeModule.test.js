import React from 'react';
import { shallow } from 'enzyme';
import EnvelopeModule from './EnvelopeModule';
import { instrumentData } from '../../constants';

const envelopeData = instrumentData.synth.envelope;

test('renders correctly', () => {
    const component = shallow(
        <EnvelopeModule 
            envelopeData={envelopeData}
            instrumentId="5542150612118159"
            additionalNesting={[]}
            handleChange={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
});