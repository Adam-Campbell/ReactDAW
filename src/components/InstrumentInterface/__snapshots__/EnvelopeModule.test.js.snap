// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="instrument-interface__module-container"
>
  <h2
    className="instrument-interface__module-heading"
  >
    Envelope
  </h2>
  <EnhancedRangeInput
    handleChange={[MockFunction]}
    identifier="5542150612118159"
    inputId="envelope-attack"
    label="Attack"
    max={1}
    min={0.005}
    propertyPathArray={
      Array [
        "envelope",
        "attack",
      ]
    }
    step={0.005}
    value={0.005}
  />
  <EnhancedSelectInput
    handleChange={[MockFunction]}
    identifier="5542150612118159"
    inputId="envelope-attack-curve"
    label="Attack Curve"
    options={
      Array [
        Object {
          "text": "Linear",
          "value": "linear",
        },
        Object {
          "text": "Exponential",
          "value": "exponential",
        },
        Object {
          "text": "Sine",
          "value": "sine",
        },
        Object {
          "text": "Cosine",
          "value": "cosine",
        },
        Object {
          "text": "Bounce",
          "value": "bounce",
        },
        Object {
          "text": "Ripple",
          "value": "ripple",
        },
        Object {
          "text": "Step",
          "value": "step",
        },
      ]
    }
    propertyPathArray={
      Array [
        "envelope",
        "attackCurve",
      ]
    }
    value="linear"
  />
  <EnhancedRangeInput
    handleChange={[MockFunction]}
    identifier="5542150612118159"
    inputId="envelope-decay"
    label="Decay"
    max={1}
    min={0.005}
    propertyPathArray={
      Array [
        "envelope",
        "decay",
      ]
    }
    step={0.005}
    value={0.1}
  />
  <EnhancedRangeInput
    handleChange={[MockFunction]}
    identifier="5542150612118159"
    inputId="envelope-release"
    label="Release"
    max={1}
    min={0.005}
    propertyPathArray={
      Array [
        "envelope",
        "release",
      ]
    }
    step={0.005}
    value={1}
  />
  <EnhancedSelectInput
    handleChange={[MockFunction]}
    identifier="5542150612118159"
    inputId="envelope-release-curve"
    label="Release Curve"
    options={
      Array [
        Object {
          "text": "Linear",
          "value": "linear",
        },
        Object {
          "text": "Exponential",
          "value": "exponential",
        },
        Object {
          "text": "Sine",
          "value": "sine",
        },
        Object {
          "text": "Cosine",
          "value": "cosine",
        },
        Object {
          "text": "Bounce",
          "value": "bounce",
        },
        Object {
          "text": "Ripple",
          "value": "ripple",
        },
        Object {
          "text": "Step",
          "value": "step",
        },
      ]
    }
    propertyPathArray={
      Array [
        "envelope",
        "releaseCurve",
      ]
    }
    value="exponential"
  />
  <EnhancedRangeInput
    handleChange={[MockFunction]}
    identifier="5542150612118159"
    inputId="envelope-sustain"
    label="Sustain"
    max={1}
    min={0}
    propertyPathArray={
      Array [
        "envelope",
        "sustain",
      ]
    }
    step={0.005}
    value={0.3}
  />
</div>
`;
